app/Services/VocabularyServiceProvider.js

```diff
@@ -97,9 +97,62 @@ async function getGroupVocabulary(userId, groupId) {
   return vocabulary;
 }
 
+async function updateVocabulary({ name, translations }, userId, vocabularyCardId) {
+  // delete all translations belonging to vocabulary card
+
+  const oldTranslations = await Translation.findAll({
+    where: {
+      userId,
+      vocabularyCardId,
+    },
+  });
+
+  await Promise.all(
+    oldTranslations.map(async (oldTranslation) => {
+      await oldTranslation.destroy();
+    })
+  );
+
+  // change name from foreign Word
+  const vocabulary = await VocabularyCard.findOne({
+    where: {
+      id: vocabularyCardId,
+      userId,
+    },
+  });
+
+  vocabulary.name = name;
+  await vocabulary.save();
+
+  // create new vocabulary cards from request
+  await Promise.all(
+    translations.map(async (translation) => {
+      await createTranslations(userId, vocabulary.languagePackageId, vocabularyCardId, translation.name);
```

Comment:
the `createTranslations` function creates translations so no need of the map and Promise.all

Reply:
Oh that was still old code that I overlooked and forgot to fix